openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/GlossaryTermsV1.component.tsx

```diff
@@ -44,13 +52,105 @@ const GlossaryTermsV1 = ({
   refreshGlossaryTerms,
   onAssetClick,
   isSummaryPanelOpen,
+  termsLoading,
+  onAddGlossaryTerm,
+  onEditGlossaryTerm,
 }: Props) => {
-  const { glossaryName: glossaryFqn } = useParams<{ glossaryName: string }>();
+  const {
+    glossaryName: glossaryFqn,
+    tab,
+    version,
+  } = useParams<{ glossaryName: string; tab: string; version: string }>();
+  const history = useHistory();
   const assetTabRef = useRef<AssetsTabRef>(null);
   const [assetModalVisible, setAssetModelVisible] = useState(false);
-
   const [assetCount, setAssetCount] = useState<number>(0);
 
+  const activeTab = useMemo(() => {
+    return tab ?? 'overview';
+  }, [tab]);
+
+  const activeTabHandler = (tab: string) => {
+    history.push({
+      pathname: version
+        ? getGlossaryTermsVersionsPath(glossaryFqn, version, tab)
+        : getGlossaryTermDetailsPath(glossaryFqn, tab),
+    });
+  };
+
+  const tabItems = useMemo(() => {
+    const items = [
+      {
+        label: <div data-testid="overview">{t('label.overview')}</div>,
+        key: 'overview',
+        children: (
+          <GlossaryOverviewTab
+            isGlossary={false}
+            permissions={permissions}
+            selectedData={glossaryTerm}
+            onUpdate={(data) => handleGlossaryTermUpdate(data as GlossaryTerm)}
+          />
+        ),
+      },
+      {
+        label: (
+          <div data-testid="terms">
+            {t('label.glossary-term-plural')}
+            <span className="p-l-xs ">
+              {getCountBadge(
+                childGlossaryTerms.length,
+                '',
+                activeTab === 'terms'
+              )}
+            </span>
+          </div>
+        ),
+        key: 'terms',
+        children: (
+          <GlossaryTermTab
+            childGlossaryTerms={childGlossaryTerms}
+            isGlossary={false}
+            permissions={permissions}
+            refreshGlossaryTerms={refreshGlossaryTerms}
+            selectedData={glossaryTerm}
+            termsLoading={termsLoading}
+            onAddGlossaryTerm={onAddGlossaryTerm}
+            onEditGlossaryTerm={onEditGlossaryTerm}
+          />
+        ),
+      },
+      {
+        label: (
+          <div data-testid="assets">
+            {t('label.asset-plural')}
+            <span className="p-l-xs ">
+              {getCountBadge(assetCount ?? 0, '', activeTab === 'assets')}
+            </span>
+          </div>
+        ),
+        key: 'assets',
+        children: (
+          <AssetsTabs
+            isSummaryPanelOpen={isSummaryPanelOpen}
+            permissions={permissions}
+            ref={assetTabRef}
+            onAddAsset={() => setAssetModelVisible(true)}
+            onAssetClick={onAssetClick}
+          />
+        ),
+      },
+    ];
+
+    return items;
+  }, [
+    glossaryTerm,
+    permissions,
+    termsLoading,
```

Comment:
We are missing many dependencies here can you double-check if those are required?
like: `onAssetClick`, `setAssetModelVisible`

Reply:
`setAssetModelVisible` is now state variable in this component and `onAssetClick` dependency is there.